WordPress/pattern-directory
# Theme: Connect `PatternGridMenu` to `PatternGrid`

This PR connects the `PatternGridMenu` to the `PatternGrid` allowing users to view patterns by categories.

I had hoped to have more atomic Pull Requests for some of the components & features introduced in this PR but found myself often blocked by missing pieces. 

### Key Updates:
- Introduces a `RouteProvider` to be used to set & follow path changes.
- Introduces a `Pattern` component to wrap `<PatternGridMenu>` & `<PatternGrid>` inside the `RouteProvider`
- Moves category-related state from the `PatternGridMenu` to the shared state introduced in #71.
- Adds a `QueryMonitor` that responds to updates to the browser's path, saves the new query in the state which triggers a new api request.


### Screenshots

_With Chrome Developer tools "Fast 3G" throttling._

![Clicking through the menu items](https://d.pr/i/6BdDyb.gif)

### How to test the changes in this Pull Request:

1. Visit `/wp-admin/edit-tags.php?taxonomy=wporg-pattern-category&post_type=wporg-pattern`.
2. Create a few categories.
3. Create a few block patterns and associate them with the categories.
4. Visit `/`.
5. Expect to see the categories on the page.
6. Click on the categories and expect to see the associated block pattern.
